src/buildstream/_frontend/app.py

```diff
@@ -909,13 +905,10 @@ def _assert_element_path(self, element_path):
     #    min_version (int): The user selected minimum BuildStream version
     #    element_path (str): The user selected element path
     #
-    def _init_project_interactive(self, project_name, min_version=None, element_path="elements"):
+    def _init_project_interactive(self, project_name, min_version, element_path):
 
         bst_major, bst_minor = utils._get_bst_api_version()
 
-        if min_version is None:
```

Comment:
This will make the `"default"` to `click.prompt()` be `None` at all times.

I think we should keep providing the default as the currently installed BuildStream version, as `click.prompt()` probably uses this to show the user what the default selection will be if they press ENTER without any value.

Reply:
Ok I understand now and I've read through the other usages of the version numbers (checking if the selected versions are "known" by installed buildstream).

What looks ambiguous to me, is the possibility for `min_version` to be `None`, which the function allows but never happens because it is always invoked with a hardcoded value.

Let's at least:
* Update `App.init_project()` docs comment for `min_version` and `element_path`, removing the notes about what the defaults are when you pass `None` here
* Same for `App.init_project_interactive()`

Ideally we should be moving `App` to use type annotations also, but we can do that separately...